### PR TITLE
Parallel run

### PR DESCRIPTION
**Pull request by PavelBal**
Implementation of running simulations concurrently.

Concurrent batch simulations will follow in a separate PR.

FYI @abdelr

### CODE REVIEW

**Comment by PavelBal** [comment]
Tests require the latest Core.

File: R/utilities-simulation.R [hunk 1]
```diff
@@ -140,6 +140,67 @@ runSimulation <- function(simulation, population = NULL, agingData = NULL, simul
...
+  for (i in seq_along(results)) {
+    resultObject <- results[[i]]
+    id <- rClr::clrGet(resultObject, "Id")
+    succeeded <- rClr::clrGet(resultObject, "Succeeded")
+    if (succeeded) {
+      # Get the correct simulation and create a SimulationResults object
+      simulationResults[[id]] <- SimulationResults$new(ref = rClr::clrGet(resultObject, "Result"), simulation = simulationsIdMap[[id]])
+      next()
+    }
+    # If the simulation run failed, show a warning
```
Comment (msevestre) [inline R/utilities-simulation.R:195]: Should not we set simulationResults[[id]] to NULL or sthg here ? to show that something is off?

**Review by msevestre — COMMENTED** [review]
I think this is looking good. Just a quick question regarding returned value in case of SIlentMode

File: R/utilities-simulation.R (hunk 1 above)
Comment (PavelBal) [reply R/utilities-simulation.R:195]: All values are NULL after initialization of `simulationResults` and are overwritten if the result exists.

**Pull request merged**